### ISSUE TITLE
OSError: [Errno 42] Protocol not available

### ISSUE THREAD

**Issue by anapaulagomes** [celery/py-amqp#193]
I got this error after updating `ampq` on MacOS (HighSierra 10.13.4).

I already tried re-installing celery 4.1 and 4.2 (rc), kombu 4.2.0, 4.1.0 amqp 2.3.1,  2.2.2 and 2.1.1.
I'm using Python 3.6.4. Rabbitmq: stable 3.7.5.

```
  File "/Users/ana/workspace/project/venv/lib/python3.6/site-packages/amqp/connection.py", line 302, in connect
    self.transport.connect()
  File "/Users/ana/workspace/project/venv/lib/python3.6/site-packages/amqp/transport.py", line 81, in connect
    self.socket_settings, self.read_timeout, self.write_timeout,
  File "/Users/ana/workspace/project/venv/lib/python3.6/site-packages/amqp/transport.py", line 166, in _init_socket
    self._set_socket_options(socket_settings)
  File "/Users/ana/workspace/project/venv/lib/python3.6/site-packages/amqp/transport.py", line 210, in _set_socket_options
    self.sock.setsockopt(SOL_TCP, opt, val)
OSError: [Errno 42] Protocol not available
```

Any help? 😕

**Comment by slallum** [comment]
I had the exact same issue. added a PR to fix it: #194

**Comment by anapaulagomes** [comment]
Nice! I've added a small comment there. I've tested here and it works for me. Thank you!
